packages/protocol/src/browser/client.ts

```diff
@@ -173,9 +174,18 @@ export class Client {
 	/**
 	 * Make a remote call for a proxy's method using proto.
 	 */
-	private remoteCall(proxyId: number | Module, method: string, args: any[]): Promise<any> {
-		if (this.disconnected) {
-			return Promise.reject(new Error("disconnected"));
+	private async remoteCall(proxyId: number | Module, method: string, args: any[]): Promise<any> {
```

Comment:
There's no `await` inside the function, and every return in this function is already a promise. It's better for either this function *not* to be `async` or return/throw regular values.

You don't need to return promises from `async` functions just for the sake of the caller getting a promise since every return will eventually be converted to a promise.

```suggestion
	private remoteCall(proxyId: number | Module, method: string, args: any[]): Promise<any> {
```

Let me know if you need more clarification on the subject. I'll be happy to expand.

Reply:
Nice catch. :+1: My best guess is that I added an `await` for something then removed it.

I also managed to forget to send the promise through to `this.parse`. Must have looked strange to see the promise put into a variable only to immediately return that variable.